File: MyNotesFront/angular-my-notes/src/app/DTOs/NoteGetDTO.ts
// src/app/models/post-dto.ts

import { Label } from "../models/label";

export interface NoteGetDTO {
  Id:string;
  Title: string;
  Content: string;
  Color: string;
  CreatedDate: Date;
  EditedDate: Date;
  IsPinned: boolean;
  IsArchived: boolean;
  UserId: string;
  GroupId: string;
  DeletedDate:Date;
  IsDeleted:boolean;
  ReminderDate:Date;
  Images:string[];
  Labels:string[];
}

File: MyNotesBackend/MyNotes.Domain/DTOs/NoteGetDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyNotes.Domain.DTOs
{
    public class NoteGetDTO
    {
        [Key]
        public string Title { get; set; }
        public string Content { get; set; }
        public string Color { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime EditedDate { get; set; }
        public bool IsPinned { get; set; }
        public bool IsArchived { get; set; }
        public long UserId { get; set; }
        public Guid GroupId { get; set; }
    }
}
